src/xmessage_builder.cpp

```diff
@@ -0,0 +1,106 @@
+#include <string>
+#include <vector>
+
+#include "nlohmann/json.hpp"
+
+#include "xeus/xmessage_builder.hpp"
+
+namespace nl = nlohmann;
+
+namespace xeus
+{
+    nl::json create_error_reply(const std::string& evalue,
```

Comment:
In Python, the error name (type) is usually first in the list of arguments, before the error value. https://docs.python.org/3/library/traceback.html#traceback.print_exception. So I would find it more natural to have the following signature:

```cpp
nl::json create_error_reply(const std::string& ename,
                            const std::string& evalue,
                            const std::vector<std::string>& trace_back)
```